app/forms/did_challenge_hmrc_form.rb

```diff
@@ -0,0 +1,3 @@
+class DidChallengeHmrcForm < BaseForm
```

Comment:
Minor thing, but this sounds stilted to me.  How about `HmrcChallenge` with a boolean attribute of `challenged`?  Or, perhaps a bit clearer `HmrcReview` with a boolean attribute of `requested`?